src/lib/services/batch-service.ts

```diff
@@ -43,55 +29,63 @@ export async function bulkTerminateByIDs({
   reason,
   workflows,
 }: CreateBatchOperationWithIDsOptions) {
-  return terminateWorkflows(namespace, queryFromWorkflows(workflows), reason);
+  const executions = workflows.map(({ runId, id }) => ({
+    runId,
+    workflowId: id,
+  }));
+  return terminateWorkflows({ namespace, reason, executions });
 }
 
 export async function batchTerminateByQuery({
   namespace,
   query,
   reason,
 }: CreateBatchOperationWithQueryOptions) {
-  return terminateWorkflows(namespace, query, reason);
+  return terminateWorkflows({ namespace, query, reason });
 }
 
 export async function bulkCancelByIDs({
   namespace,
   workflows,
   reason,
 }: CreateBatchOperationWithIDsOptions): Promise<string> {
-  return cancelWorkflows(namespace, queryFromWorkflows(workflows), reason);
+  const executions = workflows.map(({ runId, id }) => ({
```

Comment:
Wondering if we want to pull this out into a function (similar to how we had `queryFromWorkflows`)?